src/components/forms/DateInput/DateInput.tsx

```diff
@@ -0,0 +1,102 @@
+import React from 'react'
+import classnames from 'classnames'
+
+import { TextInput } from '../TextInput/TextInput'
+import { Label } from '../Label/Label'
+import { Fieldset } from '../Fieldset/Fieldset'
+
+interface DateInputElementProps {
+  id: string
+  name: string
+  label: string
+  maxLength: number
+  minLength?: number
+}
+
+interface DateInputProps {
+  id: string
+  name: string
+  legend: React.ReactNode
+  month?: boolean
+  day?: boolean
+  year?: boolean
+  hint?: string
+  className?: string
+}
+
+const DateInputElement = (
+  props: DateInputElementProps & React.HTMLAttributes<HTMLElement>
+): React.ReactElement => {
+  const { id, name, label, maxLength, minLength } = props
+
+  return (
+    <div className="usa-form-group usa-form-group--month">
+      <Label htmlFor={id}>{label}</Label>
+      <TextInput
+        className="usa-input--inline"
+        id={id}
+        name={name}
+        type="text"
+        maxLength={maxLength}
+        minLength={minLength}
+        pattern="[0-9]*"
+        inputMode="numeric"
+      />
+    </div>
+  )
+}
+
+export const DateInput = (
+  props: DateInputProps & React.HTMLAttributes<HTMLElement>
+): React.ReactElement => {
+  const {
+    id,
+    name,
+    legend,
+    month = true,
+    day = true,
+    year = true,
+    hint,
+    className,
+    ...divAttributes
+  } = props
+
+  const classes = classnames('usa-memorable-date', className)
+
+  return (
+    <Fieldset legend={legend}>
+      <span className="usa-hint" id={`${id}_hint`}>
```

Comment:
We probably want to omit this `<span>` entirely if there is no `hint`. Also, I looked around the USWDS source code and it _looks_ like a `<span className="usa-hint">` can be present in any kind of fieldset element. So I might suggest creating a `forms/Hint` component. That way users can put hint text in any fieldset, and they can also assign whatever `id` (and other span attributes) they need to. What do you think?

Reply:
In fact, looking at the USWDS markup again I think maybe this component should _just_ render the `<div class="usa-memorable-date">` and leave the fieldset/hint for the user to add. IMO unlike the header/footer, the form components should be more composable to leave them open to a variety of use cases

Reply:
Okay, I'm making one last suggestion about the structure of this component. The thing that gives me pause is that as-is, there's no way for a user to pass in custom props to the `<input>` components themselves, and this is often needed for a variety of reasons, such as hooking into a forms library or just for custom handling of events. So, at the risk of going back down the "what even are sub-components" rabbit hole, what if these components were exported separately & meant to be composed together by the user, such as...

```
<DateInputGroup> // renders <div class="usa-memorable-date"> with children
  <DateInput unit="month" /> // renders basically the DateInputElement you already have
  <DateInput unit="day" />
  <DateInput unit="year" />
</DateInputGroup>
```

... then the user can pass in custom label text, name, id, input attributes, etc. as they need, and they can even control which parts of the date are included and what order/structure they're rendered in. WDYT? I don't know if that `month/day/year` prop has to be called `unit` either, that is just what came to mind right away, but something else might be better.

Reply:
I think your last comment makes a lot of sense. I think having the DateInputGroup and DateInput components is the way to go.

Reply:
Nice. Like the idea of a new type of form group `DateInputGroup`.  I do think that the forms/hint component might also be useful generally. Maybe that's a new component to create an issue for, beyond the scope of this work.